Userland/Libraries/LibWeb/Streams/UnderlyingSink.cpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2023, Matthew Olsson <mattco@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <LibJS/Runtime/VM.h>
+#include <LibWeb/Streams/AbstractOperations.h>
+#include <LibWeb/Streams/UnderlyingSink.h>
+#include <LibWeb/WebIDL/CallbackType.h>
+
+namespace Web::Streams {
+
+JS::ThrowCompletionOr<UnderlyingSink> UnderlyingSink::from_value(JS::VM& vm, JS::Value value)
+{
+    if (!value.is_object())
+        return UnderlyingSink {};
+
+    auto& object = value.as_object();
+
+    return UnderlyingSink {
+        .start = TRY(property_to_callback(vm, value, "start")),
+        .write = TRY(property_to_callback(vm, value, "write")),
+        .close = TRY(property_to_callback(vm, value, "close")),
+        .abort = TRY(property_to_callback(vm, value, "abort")),
+        .type = TRY(object.get("type")),
```

Comment:
This still always assigns a value to the optional type. You cannot use `Object::get()` (which has an `undefined` fallback) here.